src/networking.c

```diff
@@ -2871,6 +2884,70 @@ int clientSetNameOrReply(client *c, robj *name) {
     return result;
 }
 
+/* Check validity of an attribute that's gonna be shown in CLIENT LIST. */
+int validateClientAttrOrReply(client *c, sds attr, sds val) {
+    /* Check if the charset is ok. We need to do this otherwise
+     * CLIENT LIST format will break. You should always be able to
+     * split by space to get the different fields. */
+    while (*val) {
+        if (*val < '!' || *val > '~') { /* ASCII is assumed. */
+            addReplyStatusFormat(c,
+                "%s cannot contain spaces, newlines or special characters.", attr);
+            return C_ERR;
+        }
+        val++;
+    }
+    return C_OK;
+}
+
+/* Set complementary client handshake arguments */
+void helloextCommand(client *c) {
+    if ((c->argc % 2) == 0) {
+        addReplyErrorArity(c);
+        return;
+    }
+    addReplyArrayLen(c, (c->argc-1) / 2);
+    for (int i = 1; i < c->argc; i += 2) {
+        sds attr = c->argv[i]->ptr;
+        robj *valob = c->argv[i+1];
+        sds val = valob->ptr;
+        if (!strcasecmp(attr,"LIBNAME")) {
+            if (validateClientAttrOrReply(c, attr, val)==C_ERR)
+                continue;
+            if (c->lib_name) decrRefCount(c->lib_name);
+            if (sdslen(val)) {
+                c->lib_name = valob;
+                incrRefCount(valob);
+            } else
+                c->lib_name = NULL;
+            addReply(c,shared.ok);
+        } else if (!strcasecmp(attr,"LIBVER")) {
+            if (validateClientAttrOrReply(c, attr, val)==C_ERR)
+                continue;
+            if (c->lib_ver) decrRefCount(c->lib_ver);
+            if (sdslen(val)) {
+                c->lib_ver = valob;
+                incrRefCount(valob);
+            } else
+                c->lib_ver = NULL;
+            addReply(c,shared.ok);
+        } else if (!strcasecmp(attr,"LIBENV")) {
+            if (validateClientAttrOrReply(c, attr, val)==C_ERR)
+                continue;
+            if (c->lib_env) decrRefCount(c->lib_env);
+            if (sdslen(val)) {
+                c->lib_env = valob;
+                incrRefCount(valob);
+            } else
+                c->lib_env = NULL;
+            addReply(c,shared.ok);
+        } else {
+            addReplyStatusFormat(c,"Unrecognized option '%s'", attr);
+            return;
```

Comment:
`continue`?
.e.g `helloext invalid testlibname libver 1`
`addReplyArrayLen(c, 2)` in above, but response one entry only.